### ISSUE TITLE
Use in real time

### ISSUE THREAD

**Issue by ROBERT-MCDOWELL** [salomvary/fdk-aac.js#2]
Is frame by frame or in a loop to encode raw audio into aac possible?
If yes I would love to see an example

**Comment by salomvary** [comment]
I don't know, I suggest you check out the original Fraunhofer FDK AAC library and see what it is capable of. Likely adjustments will be necessary in this repo as well.

**Comment by ROBERT-MCDOWELL** [comment]
ok thanks

**Comment by tchakabam** [comment]
@ROBERT-MCDOWELL probably yes, audio encoder libs operate at that level of granularity. in fact, there isn't much other way, the input data can only really by framed by raw audio samples.

that is, to create one AAC frame you need 1024 audio samples, so that will be the min input buffer size.

**Comment by ROBERT-MCDOWELL** [comment]
@tchakabam
thanks! if you know some simple example of how to smatrly manage the buffer and how to stream it it would be nice.

**Comment by tchakabam** [comment]
Hi @ROBERT-MCDOWELL , sorry missed this message. You can email me at stephan@emliri.com 

Your question is pretty broad, I am not sure what you really want to know at this point :)

**Comment by ROBERT-MCDOWELL** [comment]
ok thanks will do @tchakabam
I just wonder if it can be integrated into webcodecs API.

**Comment by tchakabam** [comment]
Webcodecs should already give you an encoder (if the respective browser has it). so you wouldnt need this library.

Or do you mean Webaudio?

**Comment by ROBERT-MCDOWELL** [comment]
mhmm, I thought webaudio was included in webcodecs.... very confusing... so yes webaudio.
I recently talked to a dev guys from webcodecs to integrate custom audio/video encoder decoder as plugin into the stack and he said not for now but nothing is closed about it.

**Comment by ROBERT-MCDOWELL** [comment]
here is the discussion
https://github.com/w3c/webcodecs/issues/413

**Comment by tchakabam** [comment]
> mhmm, I thought webaudio was included in webcodecs.... 

no :) those 2 are really 2 completely different stories. and there isn't any ambiguity.

webaudio is to use audio devices as in sound output, like your speakers / soundcard, or get input from microphone. (there is a lot of docs on webaudio btw, so you can just read up whatever you want to know)

webcodecs is to use a/v codecs that your platform offers, but that can be unrelated to wether you will use a soundcard.

hope that's helpful.

**Comment by tchakabam** [comment]
> I recently talked to a dev guys from webcodecs to integrate custom audio/video encoder decoder as plugin into the stack and he said not for now but nothing is closed about it.

i don't think that is really the point for webcodecs. it is mainly about offering direct access to (possibly hardware accelerated) a/v codecs that are usually platfrom native owned.

i don't see what wrapping any custom codec impl into the webcodecs api would solve itfp here.

**Comment by ROBERT-MCDOWELL** [comment]
also the biggest problem is mon threading limiting drastically the web browser. 
is webaudio open the door to integrate new audio codecs?

**Comment by ROBERT-MCDOWELL** [comment]
sorry for my question but I still do not see the difference between webcodecs and webaudio, unless webcodecs is for video and webaudio for audio?

**Comment by ROBERT-MCDOWELL** [comment]
oops I just read your answers....
ok so let's imagine an exemple, using webaudio for audio inputs and outputs, grab the raw audio/video and use webcodecs to encode in my custom codecs, attach the frames to a net stream, send it, and play it from raw data by decoding the stream with the wbcodec of my choice. This is I would like to do, removing the webrtc gas machine which the developer cannot custom anything in it.
